src/backend/bs_repserver

```diff
@@ -2524,8 +2524,12 @@ sub copybuild {
       if (defined($setrelease)) {
 	$setrelease =~ s/^-?/-/; # "-" will drop the release tag
 	$setrelease =~ s/-?$//;  # drop leading "-", it depends on the format
-	$nbin =~ s/-([^-]+)(-Media\d?(?:\..*?)?)$/$setrelease$2/; # kiwi product builds
-	$nbin =~ s/-Build[\d.]+\./$setrelease./; # kiwi appliance builds
+        # we must ensure to run only one of the following, because people
+        # overwrite -BuildX.Y part with -BuildV.W (using invalid numbers)
```

Comment:
Explanation is wrong - using any numbers whatsoever for the build is fine - as long as it is not `BuildX.Y-Media`, which matches the product because of the `-Media` and the appliance because of `BuildX.`, but the `Y-` is missing because of the `-`